src/main/java/frc/robot/swerve/SwerveDrive.java

```diff
@@ -230,18 +253,18 @@ public void updateOffsets(double[] offsets) {
      * @param fieldOriented Should the drive be field oriented.
      */
     public void drive(ChassisSpeeds chassisSpeeds, boolean fieldOriented) {
```

Comment:
A few suggestions:
1. Exit early if `chassisSpeeds` is zero -- no need to execute subsequent logic.
2. Replace the for loop inside that condition with `forEach()` which is more concise.
3. Refactoring the if-else in the end -- you duplicate `setModuleStates(kinematics.toSwerveModuleStates(...))` without need.
```java
        loggerInputs.desiredSpeeds = Utils.chassisSpeedsToArray(chassisSpeeds);

        if (chassisSpeeds.equals(new ChassisSpeeds(0, 0, 0))) {
            modules.forEach(SwerveModule::stop);
            return;
        }

        ChassisSpeeds fieldOrientedChassisSpeeds = ChassisSpeeds.fromFieldRelativeSpeeds(
                chassisSpeeds.vxMetersPerSecond,
                chassisSpeeds.vyMetersPerSecond,
                chassisSpeeds.omegaRadiansPerSecond,
                new Rotation2d(getYaw())
        );


        ChassisSpeeds targetChassisSpeeds = fieldOriented
                ? ChassisSpeeds.fromFieldRelativeSpeeds(
                chassisSpeeds.vxMetersPerSecond,
                chassisSpeeds.vyMetersPerSecond,
                chassisSpeeds.omegaRadiansPerSecond,
                new Rotation2d(getYaw())
        )
                : chassisSpeeds;
        setModuleStates(kinematics.toSwerveModuleStates(targetChassisSpeeds));
```